tensorboard/webapp/plugins/npmi/util/filter_annotations_test.ts

```diff
@@ -0,0 +1,455 @@
+
+describe('filter annotations utils', () => {
+  const annotationData: AnnotationDataListing = {
+    annotation_1: [
+      {
+        annotation: 'annotation_1',
+        metric: 'test',
+        run: 'run_1',
+        nPMIValue: 0.5178,
+        countValue: 100,
+      },
+      {
+        annotation: 'annotation_1',
+        metric: 'other',
+        run: 'run_1',
+        nPMIValue: 0.815,
+        countValue: 100,
+      },
+      {
+        annotation: 'annotation_1',
+        metric: 'test',
+        run: 'run_2',
+        nPMIValue: 0.02157,
+        countValue: 101,
+      },
+      {
+        annotation: 'annotation_1',
+        metric: 'other',
+        run: 'run_2',
+        nPMIValue: -0.02157,
+        countValue: 101,
+      },
+      {
+        annotation: 'annotation_1',
+        metric: 'test',
+        run: 'run_3',
+        nPMIValue: -0.31,
+        countValue: 53,
+      },
+      {
+        annotation: 'annotation_1',
+        metric: 'other',
+        run: 'run_3',
+        nPMIValue: -1.0,
+        countValue: 53,
+      },
+    ],
+    annotation_2: [
+      {
+        annotation: 'annotation_2',
+        metric: 'test',
+        run: 'run_1',
+        nPMIValue: null,
+        countValue: 572,
+      },
+      {
+        annotation: 'annotation_2',
+        metric: 'other',
+        run: 'run_1',
+        nPMIValue: -1.0,
+        countValue: 53,
+      },
+    ],
+    annotation_3: [
+      {
+        annotation: 'annotation_3',
+        metric: 'test',
+        run: 'run_1',
+        nPMIValue: 0.757,
+        countValue: 572,
+      },
+      {
+        annotation: 'annotation3',
+        metric: 'other',
+        run: 'run_1',
+        nPMIValue: 0.05,
+        countValue: 53,
+      },
+      {
+        annotation: 'annotation_3',
+        metric: 'test',
+        run: 'run_2',
+        nPMIValue: -0.157,
+        countValue: 572,
+      },
+      {
+        annotation: 'annotation3',
+        metric: 'other',
+        run: 'run_2',
+        nPMIValue: -0.05,
+        countValue: 53,
+      },
+    ],
+  };
+  const activeRuns = ['run_1', 'run_2'];
+  const metricArithmetic: ArithmeticElement[] = [
+    {kind: ArithmeticKind.METRIC, metric: 'nPMI@test'},
+    {kind: ArithmeticKind.OPERATOR, operator: Operator.AND},
+    {kind: ArithmeticKind.METRIC, metric: 'nPMI@other'},
+  ];
+  const metricFilters: MetricFilterListing = {
+    'nPMI@test': {
+      max: 1.0,
+      min: -1.0,
+      includeNaN: false,
+    },
+    'nPMI@other': {
+      max: 1.0,
+      min: 0,
+      includeNaN: false,
+    },
+  };
+  const metrics = ['nPMI@test', 'nPMI@other'];
+  const hiddenAnnotations = ['annotation_2', 'annotation_3'];
+
+  describe('filter annotations', () => {
+    it('returns correct result filtered by runs, active metrics, and metric filters', () => {
+      const result = filterAnnotations(
+        annotationData,
+        activeRuns,
+        metricArithmetic,
+        metricFilters,
```

Comment:
Could you add a case where the annotation is filtered away by the arithmetic operations?